src/tools/nbench/nbench.cpp

```diff
@@ -198,27 +316,35 @@ OPTIONS
                                      }
                                  },
                                  true);
+        unordered_map<string, Shape> shape_info;
         for (const string& m : models)
         {
             try
             {
                 shared_ptr<Function> f = deserialize(m);
-                cout << "Benchmarking " << m << ", " << backend << " backend, " << iterations
-                     << " iterations.\n";
-                run_benchmark(f, backend, iterations, timing_detail, warmup_iterations);
+                // cout << "Benchmarking " << m << ", " << backend << " backend, " << iterations
+                //      << " iterations.\n";
+                auto perf_data =
+                    run_benchmark(f, backend, iterations, timing_detail, warmup_iterations);
+                auto perf_shape = to_perf_shape(f, perf_data);
+                aggregate_perf_data.insert(
+                    aggregate_perf_data.end(), perf_shape.begin(), perf_shape.end());
             }
             catch (exception e)
```

Comment:
In a separate PR a check here on the exception type to discern whether it was an ngraph::unimplemented exception (for example) would be useful.